controllers/vspherevm_controller.go

```diff
@@ -602,8 +605,9 @@ func (r vmReconciler) fetchClusterModuleInfo(ctx context.Context, clusterModInpu
 		owner ctrlclient.Object
 		err   error
 	)
+	log := ctrl.LoggerFrom(ctx)
 	machine := clusterModInput.Machine
-	logger := r.Logger.WithName(machine.Namespace).WithName(machine.Name)
+	log = log.WithName(machine.Namespace).WithName(machine.Name)
```

Comment:
```suggestion
```
With the changes I have suggested above we will have the Machine as a k/v pair in this logger already